bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -133,21 +136,13 @@ private boolean isTargetBranchAllowed() {
         if (jbsIssue.isEmpty()) {
             return Optional.empty();
         }
-        org.openjdk.skara.issuetracker.Issue csr = null;
-        for (var link : jbsIssue.get().links()) {
-            var relationship = link.relationship();
-            if (relationship.isEmpty() || !relationship.get().equals("csr for")) {
-                continue;
-            }
-            csr = link.issue().orElse(null);
-            if (csr == null) {
-                log.warning("The CSR " + link + " of the issue " + issue + " does not exist");
-            } else {
-                break;
-            }
-        }
-        if (csr != null) {
-            return Issue.fromStringRelaxed(csr.id() + ": " + csr.title());
+        var csr = jbsIssue.get().links().stream()
```

Comment:
... and here as well. It's really unfortunate that we have no good framework for where to place such common functionality. :-(

Reply:
Fixed.